Cubical/Structures/Ideal.agda

```diff
@@ -0,0 +1,78 @@
+{-# OPTIONS --cubical --safe #-}
+
+module Cubical.Structures.Ideal where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Logic using ([_])
+open import Cubical.Structures.Ring
+
+private
+  variable
+    ℓ ℓ′ : Level
+
+infix 5 _∈_
+_∈_ : {X : Type ℓ } → (x : X) → (P : X → hProp ℓ′) → Type _
+x ∈ P = [ P x ]
+
+module _ (R′ : Ring {ℓ}) where
+  private
+    R = ⟨ R′ ⟩
+
+  open ring-syntax R′
+
+  {- by default, 'ideal' means two-sided ideal -}
+  record isIdeal (I : R → hProp ℓ′) : Type (ℓ-max ℓ′ ℓ) where
+    field
+      +-closed : {x y : R} → x ∈ I → y ∈ I → (x + y) ∈ I
+      -closed : {x : R} → x ∈ I → - x ∈ I
+      ₀-closed : ₀ ∈ I
```

Comment:
Why is this 0 so tiny?

Reply:
That is something I just copied from the PR on the structures (#284 ). Maybe it was used that way since 0 is not possible to use (maybe you now how that can work). I would actually prefer something like ' 0' '.

Reply:
I see... I have no clue. Maybe @Saizan knows?

Could one call it `0r` for ring zero? or `zeror`/`zero`?

Reply:
You cannot use the single digit `0` as an identifier indeed, but calling it `0r` should work, and `0-closed` should work as well.

Reply:
I will try 0r, that's easy to type and should look ok.

Reply:
Looks good to me, so it is 0r and 1r now.